### PR TITLE
OpenShift Deployment Support

### PR DESCRIPTION
**Pull request by ALinuxNinja**
- OpenShift One Click Deployment Button
- Openshift deployment hooks
- env.local.php for OpenShift
- Updating of composer.json to allow both apc/apcu

### CODE REVIEW

**Comment by jbrooksuk** [comment]
You'll need to squash this.

**Comment by StartledPhoenix** [comment]
Updated :)

**Comment by jbrooksuk** [comment]
I'm going to look through this tonight @StartledPhoenix and test it.

**Comment by tvvocold** [comment]
@StartledPhoenix

**Comment by jbrooksuk** [comment]
You'll also need to remove the `app.json` file.

**Comment by StartledPhoenix** [comment]
Done :)

**Comment by brianjking** [comment]
@jbrooksuk - Do you plan on merging this in as is? Anxious to move away from Heroku and onto OpenShift whenever you have time.

Thank you to you everyone working on Cachet & @StartledPhoenix for all the hard work!

**Comment by jbrooksuk** [comment]
Ok, some things:

- The `env.local.php` file needs to be `.env.openshift.php` (also notice the proceeding `.`)
- The `.env.openshift.php` file should include a `CACHE_DRIVER` value of `apcu` - OpenShift supports that right?

**Comment by jbrooksuk** [comment]
- What is `libs/`?
- What is `misc/`?

**Comment by StartledPhoenix** [comment]
Fixed, the two folders were left over from testing, not needed anymore

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@ vendor
 config.codekit
 !.env.heroku.php
```
Comment (jbrooksuk) [inline .gitignore]: Since you've gotten rid of the `app.json` file, can you also remove this line?

**Comment by StartledPhoenix** [comment]
Done :)

**Comment by jbrooksuk** [comment]
Thanks @StartledPhoenix - I'm going to give this a go tonight, ok? :)

**Comment by StartledPhoenix** [comment]
Great!

**Comment by tvvocold** [comment]
@jbrooksuk 
Nice! But i think maybe you should merge fisrt? 
( Right? @StartledPhoenix (Or create a new branch to test out?))

**Comment by StartledPhoenix** [comment]
The url in the button needs to be pointed to the correct git repo and branch. It can be manually adjusted in the openshift deployment page after clicking the deploy button or visiting the URL.

**Comment by brianjking** [comment]
@jbrooksuk & @StartledPhoenix I was able to deploy using the OpenShift button @StartledPhoenix created by changing the Git source repo to be that of https://github.com/StartledPhoenix/openshift-cachet and was able to deploy. Hoping this gets merged in and I can move away from Heroku! 

Thanks!

File: deplist.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (jbrooksuk) [inline deplist.txt]: Do we need this file?

**Comment by StartledPhoenix** [comment]
Fixed

**Comment by brianjking** [comment]
@StartledPhoenix & @jbrooksuk -- Anything I can do to help move this along from a testing side of things? Thanks! :smile:

**Comment by jbrooksuk** [comment]
I need to tag the current `master` as a Heroku friendly release first :smile:

**Comment by StartledPhoenix** [comment]
L5 is working as well, will be adding support in a few days.

**Comment by brianjking** [comment]
Is this no longer being developed?

**Comment by jbrooksuk** [comment]
I think it's because of the branch changes.

**Comment by brianjking** [comment]
@alinuxninja

**Pull request closed without merging**